Content.Server/_Harmony/GameTicking/Rules/ConspiratorRuleSystem.cs

```diff
@@ -0,0 +1,98 @@
+{
+
+    [Dependency] private readonly AntagSelectionSystem _antag = default!;
+    [Dependency] private readonly SharedMindSystem _mind = default!;
+    [Dependency] private readonly SharedObjectivesSystem _objectives = default!;
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ConspiratorRoleComponent, GetBriefingEvent>(OnGetBriefing);
+        SubscribeLocalEvent<ConspiratorRuleComponent, AfterAntagEntitySelectedEvent>(OnAntagSelected);
+    }
+
+    protected override void AppendRoundEndText(EntityUid uid,
+        ConspiratorRuleComponent component,
+        GameRuleComponent gameRule,
+        ref RoundEndTextAppendEvent args)
+    {
+        base.AppendRoundEndText(uid, component, gameRule, ref args);
+
+        var sessionData = _antag.GetAntagIdentifiers(uid);
+        args.AddLine(Loc.GetString("conspirator-count", ("count", sessionData.Count)));
+        foreach (var (mind, data, name) in sessionData)
+        {
+            args.AddLine(Loc.GetString("conspirator-name-user",
+                ("name", name),
+                ("username", data.UserName)));
+        }
+        if (!_proto.TryIndex(component.Objective, out var objectiveProto)) return;
+
+        args.AddLine(Loc.GetString("conspirator-objective", ("objective", objectiveProto.Name)));
+    }
+
+    private void OnGetBriefing(EntityUid uid, ConspiratorRoleComponent comp, ref GetBriefingEvent args)
+    {
+        args.Append(Loc.GetString("conspirator-identities"));
+
+        var conspirators = AllEntityQuery<ConspiratorComponent>();
+        while (conspirators.MoveNext(out var id, out _))
+        {
+            args.Append(Loc.GetString("conspirator-name", ("name", Name(id))));
+        }
+
+        args.Append(Loc.GetString("conspirator-radio-implant"));
+    }
+
+    private void OnAntagSelected(Entity<ConspiratorRuleComponent> ent, ref AfterAntagEntitySelectedEvent args)
+    {
+        if (!_mind.TryGetMind(args.Session, out var mindId, out var mind))
+            return;
+
+        if (ent.Comp.Objective is null)
+        {
+            if (GetRandomObjectivePrototype(ent.Comp, out var objectiveProtoId))
+                ent.Comp.Objective = objectiveProtoId;
+        }
+
+        if (ent.Comp.Objective is not null)
+            _mind.TryAddObjective(mindId, mind, ent.Comp.Objective);
+        else
+            return;
+    }
+
+    private bool GetRandomObjectivePrototype(ConspiratorRuleComponent comp, out EntProtoId? objectiveProto)
+    {
+        objectiveProto = null;
+        if (!_proto.TryIndex(comp.ObjectiveGroup, out var group)) return false;
+        var objectives = group.Weights.ShallowClone();
+        while (_random.TryPickAndTake(objectives, out var proto))
+        {
+            objectiveProto = proto;
```

Comment:
If you're going to be returning a boolean success value then you should also add the proper attribute to tell the compiler that the value is not null when the output is true
```suggestion
    private bool GetRandomObjectivePrototype(ConspiratorRuleComponent comp, [NotNullWhen(true)] out EntProtoId? objectiveProto)
    {
        objectiveProto = null;
        if (!_proto.TryIndex(comp.ObjectiveGroup, out var group)) return false;
        var objectives = group.Weights.ShallowClone();
        while (_random.TryPickAndTake(objectives, out var proto))
        {
            objectiveProto = proto!;
```